flink-python/pyflink/fn_execution/embedded/converters.py

```diff
@@ -214,9 +228,23 @@ def from_field_type_proto(field_type):
             [from_field_type_proto(f.type) for f in field_type.row_schema.fields],
             [f.name for f in field_type.row_schema.fields])
     elif type_name == schema_type_name.BASIC_ARRAY:
-        return ListDataConverter(from_field_type_proto(field_type.collection_element_type))
+        return ArrayDataConverter(from_field_type_proto(field_type.collection_element_type))
     elif type_name == schema_type_name.MAP:
         return DictDataConverter(from_field_type_proto(field_type.map_info.key_type),
                                  from_field_type_proto(field_type.map_info.value_type))
 
     return IdentityDataConverter()
+
+
+def from_type_info(type_info: TypeInformation):
+    if isinstance(type_info, (PickledBytesTypeInfo, RowTypeInfo, TupleTypeInfo)):
```

Comment:
Because pemja doesn't recognize these two types, which means that we need data converters to transform these state data. I think we can use this as an optimization later. WDYT

Reply:
As long as users not use state processor to read it, or the state type might be a confusing point. Since we have the same issue in process mode, I think it's okay for now.